src/pages/index.js

```diff
@@ -1,75 +1,193 @@
-   popupWithFormProfile.open();
- });
-
- addNewCardButton.addEventListener("click", () => {
-   popupWithFormAddCard.open();
- });
+import Api from "../components/Api.js";
+import { enableValidation, formValidators } from "../components/FormValidator.js";
+import { validationSettings } from "../utils/Constants.js";
+
+// Elements
+const profileEditButton = document.querySelector("#profile-edit-button");
+const addNewCardButton = document.querySelector(".profile__add-button");
+const avatarEditButton = document.querySelector(".profile__image-edit-button");
+
+// API Instance
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "703aeebf-8509-4c4b-9691-d3d6d590ee64",
+    "Content-Type": "application/json"
+  }
+});
+
+// UserInfo instance
+const userInfo = new UserInfo({
+  nameSelector: ".profile__title",
+  descriptionSelector: ".profile__description",
+  avatarSelector: ".profile__image"
+});
+
+// Popup instances
+const popupWithImage = new PopupWithImage("#modal-preview-picture");
+popupWithImage.setEventListeners();
+
+const popupWithFormProfile = new PopupWithForm("#profile-edit-modal", handleProfileFormSubmit);
+popupWithFormProfile.setEventListeners();
+
+const popupWithFormAddCard = new PopupWithForm("#add-card-modal", handleAddCardFormSubmit);
+popupWithFormAddCard.setEventListeners();
+
+const popupWithFormAvatar = new PopupWithForm("#edit-avatar-modal", handleAvatarFormSubmit);
+popupWithFormAvatar.setEventListeners();
+
+const popupWithConfirmation = new PopupWithConfirmation({
+  popupSelector: '#delete-confirmation-modal',
+  handleConfirm: handleDeleteCard
+});
+popupWithConfirmation.setEventListeners();
+
+// Card section instance
+const cardSection = new Section(
+  {
+    renderer: (cardData) => {
+      const cardElement = createCard(cardData);
+      cardSection.addItem(cardElement);
+    }
+  },
+  ".cards__list"
+);
+
+// Load user information and initial cards
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([userData, initialCards]) => {
+    userInfo.setUserInfo({
+      name: userData.name,
+      description: userData.about,
+      avatar: userData.avatar,
+    });
+    cardSection.renderItems(initialCards);
+  })
+  .catch((err) => {
+    console.error("Error loading initial data:", err);
+  });
+
+// Handle profile form submission
+function handleProfileFormSubmit(formData) {
+  popupWithFormProfile.renderLoading(true); 
+  return api.updateUserInfo(formData)
+    .then((updatedData) => {
+      userInfo.setUserInfo({
+        name: updatedData.name,
+        description: updatedData.about,
+        avatar: updatedData.avatar
+      });
+      popupWithFormProfile.close();
+      popupWithFormProfile._form.reset();  // Clear the form fields only after submission
+
+      // Use toggleButtonState instead of manually disabling the button
+      formValidators['profile-form']._toggleButtonState();
```

Comment:
Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.

**Please, correct it everywhere in the code**